### DISCUSSION TITLE
CloudlogCAT / Interface to UDP-Messages

### DISCUSSION THREAD

**Discussion by dk5cr** [magicbug/Cloudlog#2786] [Ideas / Feature Requests]
Currently i've got the problem in running wsjtx (2.6.1.) and CloudlogCAT on a local windows client and Cloudlog on a remote linux-system. 

When starting CloudlogCAT in local windows 10, it sets its communication via OmniRig to COM8 and interacts with my Icom 7300, with the correct API it pushes the data to cloudlog. So far so good.
BUT: Wsjtx also tries to use COM8 to communicate with the rig. Depending on what program starts first, either WSJTX or CloudlogCAT reports an error, that COM8 is occupied. 
WSJTX has the possibility to get its data from device "Hamlib NET rigctl" via ip:4532 . 
Otherwise WSJTX can also push the current information, received from the rig via USB, to port 4532. 

So it would be great, if CloudlogCAT gets the possibility to connect to Port 4532 and listens to the data for beaming it to cloudlog without connecting via OmniRig on COM8 or so. 

Or do i have a thinking problem in finding a solution to get these three programms running with data exchange between WSJTX, Cloudlog and - if necessary - CloudlogCAT?

**Comment by phl0** [comment]
This should better be placed in https://github.com/rmc47/CloudlogCAT repo?!

**Reply by phl0** [reply]
Oopps looks like it is archived ...

**Reply by magicbug** [reply]
Nope its been a long time since CloudlogCAT has been the codebase of that repo, I've no plan currently to add hamlib support to CloudlogCAT however it could be something that's included in Cloudlog Aurora which will ultimately replace it.

Worth saying that I believe WSJT-X does support Omnirig which allows more than one app to access the same radio

**Comment by phl0** [comment]
But why do you need Omnirig? WSJT-X can directly CAT control your TRX. And there is WSJT-X improved which includes a patch by me to directly log to Cloudlog. See https://sourceforge.net/projects/wsjt-x-improved/files/WSJT-X_v2.7.1/.

**Reply by HB9HIL** [reply]
I never used that before 😆 I usually used Aurora to log the qso.

**Comment by dk5cr** [comment]
Wonderful, thanks for this hint. I just installed 2.7.1. devel, added URL and API key. 
But cloudlog doesn't recognize the information coming from WSJTX. 
How do i install or activate this "hardware" interface in cloudlog? 

The API-test worked well and showed a green confirmation.

**Reply by HB9HIL** [reply]
https://github.com/magicbug/Cloudlog/wiki/WSJT-X-Integration

**Reply by HB9HIL** [reply]
https://github.com/magicbug/Cloudlog/wiki/Radio-Interface

**Comment by dk5cr** [comment]
I just did exact what is described in the links, adding URL and API-Key, checking station-id. It fits. 
Unfortunately there is no hint what to set in "Radio-Interface" in Cloudlog to get it working... 
In the list of the hardware interfaces my wsjtx doesn't appear...

**Reply by HB9HIL** [reply]
You can check if there are coming some data in Cloudlog in the Hardware Interfaces Menu

<img width="341" alt="hardware interfaces" src="https://github.com/magicbug/Cloudlog/assets/80885850/834538b5-39d7-484b-a2b0-93bd81aeea5c">

If yes then you can fetch this data in the qso windows by choosing the corresponding radio in the tabbed menu

<img width="578" alt="radio" src="https://github.com/magicbug/Cloudlog/assets/80885850/b74fd9da-a730-4e89-8cae-4a2da4a67726">

**Comment by dk5cr** [comment]
Sorry, but... 

Test API key in WSJTX-271-dev is green, so i assume, that the data arrives at cloudlog. But Hardware-Interfaces doesn't list anything else than the "old" OmniRig1 and 2. 

Is there a chance to see or check, what cloudlog receives? 
What name should appear in Cloudlog for received "Hardware info" from WSJTX? 

**[add]** I deleted the old OmniRig1 and 2 to prevent showing old entries will hide new entries... but "No CAT Interfaced radios found."

**[add2]** I restarted CloudlogCAT and when clicking on Hardware interfaces in Cloudlog the OmniRig1 appears after 1 sec., so connection between Windows client and linux server works properly.

**[add3]** Just installed wsjtx271-dev on my Linux standalone and tried to get it working in cloudlog on remote-linux on my raspberry/BananaPi. But it showed the same error or better non-function. No hardware interface in cloudlog found :-(

**[add4]** TCPdump on my linux-system logged a whole bunch of data, coming from my Windows-PC. Grepping for cloudlog a list of GET-requests will be shown, send from Windows to linux. So it seems, that wsjtx271dev works properly, but cloudlog couldn't handle the requests. 

shortened and cleanedup this is approaching my linux box (id changed to ....... for security issues and ip-address replaced by 192.168.x.xx.) 
"Referer: http://192.168.x.xx/cloudlog/index.php/radio
Cookie: radio=7; station_profile_id=1; ci_cloudlog=fdi6........0mc8lr
        Set-Cookie: ci_cloudlog=fdi6........0mc8lr; path=/; HttpOnly
Set-Cookie: ci_cloudlog=fdi6........0mc8lr; path=/; HttpOnly
        GET /cloudlog/index.php/radio/status/ HTTP/1.1"

**Comment by dk5cr** [comment]
Ok, automatic loggig from WSJTX2.7.1.dev works. When logging QSO in WSJTX in the seperate window, a second later it's in cloudlog, when refreshing the screen. But having data an band, frequency, lsb/usb and so on is still missing, because no hardware interface is found in cloudlog.

**Reply by bgritzmacher** [reply]
That's because integration with WSJT-X *only* does logging, it doesn't push live radio information, such as frequency, mode, etc. As such, you'll never see a "rig" associated with WSJT-X.

**Reply by HB9HIL** [reply]
The QSO already containes necessary band(qrg)/mode information

**Reply by dk5cr** [reply]
Thanks for your informationen. But my question at first was, how to adjust my rig when changing WSJTX and the answer was: 
Why don't i use wsjtx271dev and HB9HIL attached a screenshot with hardware interfaces in Cloudlog fed by WSJTX271. 

So the question is still alive: 
OmniRig and WSJTX (and others) won't work together because they use the same interface (COM8 in my case)...

**Reply by HB9HIL** [reply]
Yeah seems like there was a misunderstandment:

If you work WSJT-X you log QSO directly (i never used that) or with Cloudlog Aurora which pushes the QSO with an API Key directly to Cloudlog. You don't have the classic hardware interface then because the QSO already have the band, frequency and mode information. 

If you work normally on the bands or with a software which do not support logging then you can use the hardware interface with omnirig and cloudlogcat to push radio information to Cloudlog. 

Seems like you wanted to mix those two cases. 

> So the question is still alive:
OmniRig and WSJTX (and others) won't work together because they use the same interface (COM8 in my case)...

To answer your question: You don't use omnirig and WSJT-X together because WSJT-X is controlling your radio and pushing only the QSO to Cloudlog. You don't need Omnirig and the Hardware Interface Feature of Cloudlog in the case of WSJT-X

**Reply by HB9HIL** [reply]
@dk5cr You can mail me if you want, Mail is on QRZ.com. Then I can help you in german 😉

**Reply by dk5cr** [reply]
Ok thanks. Seems like i mixed it up. But anyway logging works now properly and due to the hint of the CloudlogCAT-developper i'm now aware, that the new WSJTX also has an OmniRig-Selection in the rig's list. 
Thanks for great support and sorry for my misunderstanding. 
cheers and goodDX 
Cornel DK5CR

**Reply by HB9HIL** [reply]
You're welcome. If your question is finally answered, please mark this discussion as answered 😄 thank you
